src/js/components/Widgets/AnnotatedSlideshow.jsx

```diff
@@ -23,14 +27,27 @@ class AnnotatedSlideshow extends PureComponent {
     }
     // this.handleSlideImage(num);
     this.props.onChangeSlide(num ? selectedStepIndex + 1 : selectedStepIndex - 1);
+    this.autoAdvanceSlide();
+  }
+
+  autoAdvanceSlide () {
+    clearTimeout(this.timer);
+    const { slides, selectedStepIndex} = this.props;
+    const data = Object.values(slides);
+    // const { length } = data;
+    const { delayBeforeAdvancingSlide } = data.find(slide => slide.index === selectedStepIndex);
+    this.timer = setTimeout(() => {
+      this.handleChangeSlide(1);
+    }, delayBeforeAdvancingSlide);
+
   }
 
   render () {
     const { slides, selectedStepIndex, classes } = this.props;
     const data = Object.values(slides);
     const { length } = data;
     const { title, description, imgSrc } = data.find(slide => slide.index === selectedStepIndex);
-
+    this.componentDidMount();
```

Comment:
Hi @LohitG componentDidMount is a built in lifecycle function that shouldn’t be called with every render.